model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -179,10 +180,17 @@ object TagInliner extends Logging {
         }
 
       case AssignmentTag(value, assignTo) =>
-        for {
-          cd <- valueToModel(value)
-          _ <- Exports[S].resolved(assignTo, cd._1)
-        } yield Some(SeqModel) -> cd._2
+        (value match {
+          // if we assign collection to a stream, we must use it's name, because it is already created with 'new'
+          case c@CollectionRaw(_, _: StreamType) =>
+            collectionToModel(c, Some(assignTo))
```

Comment:
Do you ever call `collectionToModel(..., None)`?

Reply:
yes, in unfold